ui/components/logs/ui/code-editor.tsx

```
@@ -0,0 +1,492 @@
+					}),
+				);
+			});
+
+			// Register completion provider for non-object items
+			newDisposables.push(
+				monaco.languages.registerCompletionItemProvider(lang || "javascript", {
+					provideCompletionItems: (model, position) => {
+						const wordUntilPosition = model.getWordUntilPosition(position);
+						return {
+							suggestions: props
+								.customCompletions!.filter((item) => item.type !== "object")
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)